403 forbidden - invalid package

I am attempting to download a package hosted on npmjs.org using my Verdaccio installation as a mirror.  This works for many other packages I have used.  The package in question is here:

https://www.npmjs.com/package/@ember-data/-build-infra

I suspect the problem is the leading dash in the name.  When I attempt to install through Verdaccio, I get a 403 invalid package error.  The expected behavior is that the package successfully installs.

In the console:

```
$ npm install @ember-data/-build-infra
npm ERR! code E403
npm ERR! 403 Forbidden - GET https://<<HOSTNAME>>/@ember-data%2f-build-infra - invalid package
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/deploy/.npm/_logs/2019-07-22T21_34_35_025Z-debug.log
[deploy@ip-172-31-26-34 ~]$ npm install @ember-data/-build-infra
npm ERR! code E403
npm ERR! 403 Forbidden - GET https://npm-registry.power.auction/@ember-data%2f-build-infra - invalid package

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/deploy/.npm/_logs/2019-07-22T21_34_41_594Z-debug.log
```

In the verdaccio log:

```
{"name":"verdaccio","hostname":"<<HOSTNAME>>","pid":29999,"sub":"in","level":30,"req":{"method":"GET","url":"/@ember-data%2f-build-infra","headers":{"host":"npm-registry.power.auction","x-forwarded-for":"34.228.153.10","x-forwarded-proto":"https","connection":"close","user-agent":"npm/6.9.0 node/v10.16.0 linux x64","npm-in-ci":"false","npm-scope":"","npm-session":"9279a6842df72517","referer":"install [REDACTED]","pacote-req-type":"packument","pacote-pkg-id":"registry:@ember-data/-build-infra","accept":"application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*","authorization":"<Classified>","accept-encoding":"gzip,deflate"},"remoteAddress":"127.0.0.1","remotePort":44598},"ip":"127.0.0.1","msg":"127.0.0.1 requested 'GET /@ember-data%2f-build-infra'","time":"2019-07-22T21:38:55.442Z","v":0}
{"name":"verdaccio","hostname":"<<HOSTNAME>>","pid":29999,"sub":"in","level":35,"request":{"method":"GET","url":"/@ember-data%2f-build-infra"},"user":null,"remoteIP":"34.228.153.10 via 127.0.0.1","status":403,"error":"invalid package","bytes":{"in":0,"out":33},"msg":"403, user: null(34.228.153.10 via 127.0.0.1), req: 'GET /@ember-data%2f-build-infra', error: invalid package","time":"2019-07-22T21:38:55.444Z","v":0}
```

> Environment Info:
> 
>   System:
>     OS: Linux 3.10 Red Hat Enterprise Linux Server 7.6 (Maipo)
>     CPU: (32) x64 Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz
>   Binaries:
>     Node: 10.16.0 - /usr/bin/node
>     Yarn: 1.17.3 - /usr/bin/yarn
>     npm: 6.9.0 - /usr/bin/npm
>   Virtualization:
>     Docker: 18.09.8 - /usr/bin/docker
>   Browsers:
>     Chrome: 75.0.3770.142

--- a/src/lib/utils.ts
+++ b/src/lib/utils.ts
@@ -50,7 +50,7 @@ export function convertPayloadToBase64(payload: string): Buffer {
  * @param {*} name  the package name
  * @return {Boolean} whether is valid or not
  */
-export function validateName(name: string, isScoped: boolean = false): boolean {
+export function validateName(name: string): boolean {
   if (_.isString(name) === false) {
     return false;
   }

--- a/src/lib/utils.ts
+++ b/src/lib/utils.ts
@@ -50,18 +50,27 @@ export function convertPayloadToBase64(payload: string): Buffer {
  * @param {*} name  the package name
  * @return {Boolean} whether is valid or not
  */
-export function validateName(name: string): boolean {
+export function validateName(name: string, isScoped: boolean = false): boolean {
   if (_.isString(name) === false) {
     return false;
   }
 
   const normalizedName: string = name.toLowerCase();
 
-  // all URL-safe characters and "@" for issue #75
+  /**
+   * Some context about the first regex
+   * - npm used to have a different tarball naming system.
+   * eg: http://registry.npmjs.com/thirty-two
+   * https://registry.npmjs.org/thirty-two/-/thirty-two@0.0.1.tgz
+   * The file name thirty-two@0.0.1.tgz, the version and the pkg name was separated by an at (@)
+   * while nowadays the naming system is based in dashes
+   * https://registry.npmjs.org/verdaccio/-/verdaccio-1.4.0.tgz
+   *
+   * more info here: https://github.com/rlidwka/sinopia/issues/75
+   */
   return !(
     !normalizedName.match(/^[-a-zA-Z0-9_.!~*'()@]+$/) ||
     normalizedName.charAt(0) === '.' || // ".bin", etc.
-    normalizedName.charAt(0) === '-' || // "-" is reserved by couchdb
     normalizedName === 'node_modules' ||
     normalizedName === '__proto__' ||
     normalizedName === 'favicon.ico'

--- a/test/unit/modules/utils/utils.spec.ts
+++ b/test/unit/modules/utils/utils.spec.ts
@@ -246,6 +246,8 @@ describe('Utilities', () => {
           expect(validateName('verdaccio')).toBeTruthy();
           expect(validateName('some.weird.package-zzz')).toBeTruthy();
           expect(validateName('old-package@0.1.2.tgz')).toBeTruthy();
+          // fix https://github.com/verdaccio/verdaccio/issues/1400
+          expect(validateName('-build-infra')).toBeTruthy();
         });
 
         test('should be valid using uppercase', () => {

